torch/testing/_internal/common_utils.py

```diff
@@ -1348,13 +1349,22 @@ def disable_functorch():
         del guard
 
 
+@contextlib.contextmanager
+def _no_torch_function_mode():
+    if _is_torch_function_mode_enabled():
+        with _pop_mode_temporarily():
+            yield
+    else:
+        yield
```

Comment:
`_no_torch_function_mode` was removed in https://github.com/pytorch/pytorch/pull/86458, so I define my own helper. In that PR, see `handle_torch_function` for how the new API is used. `_pop_mode_temporarily` has no check for whether the stack is empty, only an assert on the CPP side, so I check if the mode is enabled to avoid popping an empty stack.